gutenberg/acquire/metadata.py

```diff
@@ -26,23 +28,148 @@
 from gutenberg._util.os import remove
 
 
-_METADATA_CACHE = local_path(os.path.join('metadata', 'metadata.db'))
-_METADATA_DATABASE_SINGLETON = None
-
-
-@contextlib.contextmanager
-def _download_metadata_archive():
-    """Makes a remote call to the Project Gutenberg servers and downloads the
-    entire Project Gutenberg meta-data catalog. The catalog describes the texts
-    on Project Gutenberg in RDF. The function returns a file-pointer to the
-    catalog.
-
-    """
-    data_url = r'http://www.gutenberg.org/cache/epub/feeds/rdf-files.tar.bz2'
-    with tempfile.NamedTemporaryFile(delete=False) as metadata_archive:
-        shutil.copyfileobj(urllib2.urlopen(data_url), metadata_archive)
-    yield metadata_archive.name
-    remove(metadata_archive.name)
+_GUTENBERG_CATALOG_URL = \
+    r'http://www.gutenberg.org/cache/epub/feeds/rdf-files.tar.bz2'
+
+
+class CacheAlreadyExistsException(Exception):
+    pass
+
+class CacheDeleteException(Exception):
+    pass
+
+class InvalidCacheException(Exception):
+    pass
+
+class MetadataCacheManager(object):
+    def __init__(self, store, cache_uri):
+        self.identifier='urn:gutenberg:metadata'
+        if store == 'Sleepycat':
+            self.store = store
+            self.removable = True
+        else:
+            if cache_uri.startswith('sqlite://'):
+                self.removable = True
+            else:
+                self.removable = False
```

Comment:
@cpeel: Is support for non-sqlite/non-removable graphs something on which you depend? There are no tests covering this code branch so I'd assume that this is not a feature that you are using and can therefore safely be removed. Could you please confirm this?